### PR TITLE
chore: convert components to ts

### PR DESCRIPTION
**Pull request by alextranwork**
### 📃 Summary
Refactor code to provide typings for the Webapp and EMR

### 🔍 Changeset
- Everything should behave the same except for autofucus for inputs.  Its been changed to autoFocus

### 🏷 Asana Task
https://app.asana.com/0/0/1186084579231283/f

### 🧪 QA
- Need to regression test Webapp and EMR
	
### 📈 Risk Analysis
- [ ] Does this affect patients?
- [ ] Does this affect clinical?
- [ ] Does this mutate data?
- [ ] Does it touch PHI?
- [ ] Does it touch payments?
- [ ] Does it update Node packages?

### CODE REVIEW

**Comment by vietnogigoodrx** [comment]
@rkrishn7 I reverted your merge because there were too many TS errors.  Can you fix the TS error and try merging again?  We can try merging smaller pieces if that help

**Comment by rkrishn7** [comment]
> @rkrishn7 I reverted your merge because there were too many TS errors. Can you fix the TS errors and try merging again? We can try merging smaller pieces if that help. Also, you can run `yarn types` to check for errors.

my bad, forgot to cleanup some of the tests/stories. There are a few remaining errors that are coming from `Flex` and `Input`. The ones from `Flex` should be resolved with your changes. Did you convert Input? If so, wanna take a look at that one.

There are a few couple errors from `src/Card/Card.tsx`. Specifically:

```ts
& + ${Card.Body} {
```

Not sure why we use this.

**Comment by vietnogigoodrx** [comment]
@rkrishn7 lets create PRs to merge onto this branch.  I would like to keep it error free to make it easier.  Can you revert your last change since it caused errors again.

File: src/Alert/Alert.tsx [hunk 1]
```diff
@@ -1,21 +1,25 @@
 import React from 'react';
-import PropTypes from 'prop-types';
 import { css } from 'styled-components';
-import { space } from 'styled-system';
-import propTypes from '@styled-system/prop-types';
+import { space, SpaceProps } from 'styled-system';
 import { getComponentVariant, createComponent } from '../utils';
 
-const StyledAlert = createComponent({
+export interface AlertProps extends SpaceProps {
```
Comment (schlegz) [inline src/Alert/Alert.tsx:6]: Looking through a couple related PRs, I'm seeing discrepancies around defining props.  Here it is an interface, elsewhere I've seen type.

Comment (vietnogigoodrx) [reply src/Alert/Alert.tsx:6]: hmm, I normally use interface to define objects.  Is there a reason why we would use type over interfaces for objects?

**Review by choochootrain — APPROVED** [review]
🚀

**Pull request merged**